x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/pivot/DateHistorgramFieldCollectorTests.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.transform.transforms.pivot;
+
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.action.search.SearchResponseSections;
+import org.elasticsearch.action.search.ShardSearchFailure;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.search.aggregations.Aggregations;
+import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
+import org.elasticsearch.search.aggregations.metrics.NumericMetricsAggregation;
+import org.elasticsearch.search.aggregations.metrics.NumericMetricsAggregation.SingleValue;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.core.transform.transforms.TransformCheckpoint;
+import org.elasticsearch.xpack.core.transform.transforms.pivot.DateHistogramGroupSource;
+import org.elasticsearch.xpack.core.transform.transforms.pivot.SingleGroupSource;
+import org.elasticsearch.xpack.transform.transforms.Function.ChangeCollector;
+import org.junit.Before;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DateHistorgramFieldCollectorTests extends ESTestCase {
+    Map<String, SingleGroupSource> groups = new LinkedHashMap<>();
+
+    SingleValue minTimestamp = mock(NumericMetricsAggregation.SingleValue.class);
+    SingleValue maxTimestamp = mock(NumericMetricsAggregation.SingleValue.class);
```

Comment:
a mix of per suite and per test (`@Before`) initialization is unclean. Apart from that, members should be `private`. I think it is cleaner if every test is independent and uses own mocks. You can still re-use code. My suggestion is to introduce but not initialize the variable:

```
private SingleValue minTimestamp;
private SingleValue maxTimestamp;
```

and set it up in 

```
    @Before
    public void setupDateHistogramFieldCollectorTest() {
         minTimestamp = mock(NumericMetricsAggregation.SingleValue.class);
         maxTimestamp = mock(NumericMetricsAggregation.SingleValue.class);
        ...
```